### PR TITLE
Check to see if we're running in GNOME Shell and if so what version in Notification Icon Plugin

### PR DESCRIPTION
**Pull request by JasonLG1979**
Tray icons are not supported in GNOME Shell 3.26 +. This disable the notification icon plugin if we are running in GNOME Shell version 3.26+ and don't have appindicator support and tell the user why it was disabled.

### CODE REVIEW

File: pithos/plugins/notification_icon.py [hunk 1]
```diff
@@ -91,7 +111,18 @@ def on_has_name_owner(bus, result):
                 self.prepare_complete()
             else:
                 # No fancy tray here
-                self.prepare_complete()
+                logging.info('Checking if org.gnome.Shell is owned and version')
+                Gio.DBusProxy.new(
+                    self.bus,
+                    Gio.DBusProxyFlags.NONE,
+                    None,
+                    'org.gnome.Shell',
```
Comment (TingPing) [inline pithos/plugins/notification_icon.py]: I think we discussed this before, but was there really no better way to find this information? I really don't want to open such as huge hole in the Flatpak sandbox to talk to `org.gnome.Shell`.

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: No, not that I can tell beyond sniffing the DE in python itself which is not reliable and I'm not sure how that would even work in Flatpack?

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: The other time this came up was with the notify plugin. In that case the notification interface itself returns a version.

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: The only real alternatives are either just skip the check and always assume one way or the other or just completely disable the notification icon plugin altogether.

Comment (TingPing) [inline pithos/plugins/notification_icon.py]: Hmm how about this:

```python
if os.environ['XDG_CURRENT_DESKTOP'] == 'GNOME':
     # Check org.gnome.Shell for version, if fails assume not supported
else:
     # Previous behavior
```

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: Last time I checked several DE's reported GNOME as the DE in

```python
os.environ['XDG_CURRENT_DESKTOP']
```
As I said sniffing DE's in python isn't reliable.

Comment (TingPing) [inline pithos/plugins/notification_icon.py]: oh =(

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: Just google 'XDG_CURRENT_DESKTOP' and you'll see a billion bug reports of apps using that to sniff the DE and failing.

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: Can you not just open up flatpak just enough to read that prop and that's all?

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: I'm surprised that GNOME Shell doesn't expose that to Flatpak by default. You'd think that would be nice thing to know?

Comment (TingPing) [inline pithos/plugins/notification_icon.py]: Currently the Flatpak dbus support is based upon bus names only. I would love to lock it down by property or interface but that requires work there.

Comment (JasonLG1979) [inline pithos/plugins/notification_icon.py]: Looks like you got some work to do then ;P

In my mind you'd want to be able to lock it down to the point that you could even specify if the app is even allowed to set read/write props.

**Comment by JasonLG1979** [comment]
Well no need for this anymore ;)

**Pull request closed without merging**